docs/testing/test-types/README.md

```diff
@@ -0,0 +1,46 @@
+# Test Types
+
+In the LocalStack codebase we differentiate between the following test types:
+
+- Unit tests
+- Acceptance Tests
+- Integration Tests
+
+Depending on the workflow and its trigger not all of those tests are executed at once.
+For ordinary pushes to `master` we only want to execute the Unit and Acceptance tests.
+On a regular schedule, however, we want to execute all tests to have as big of a coverage of our logic as possible.
+This differentiation also educates what we expect from the different types of tests.
+
+## Unit tests
+
+As the name suggests, these tests are performed on smaller units of logic to check if they're sound and perform the operations they claim to.
+This small unit can most often be a kind of helper function inside of a larger procedure.
+These tests should be able to complete their execution very quickly, so they never contain any interaction with some kind of infrastructure.
+If you need some kind of waiting mechanism in your unit test, it is most likely that you are not writing a unit test.
+
+
+## Acceptance tests
+
+We use acceptance tests to gain a quick understanding of whether the recently pushed commit to `master` fulfils minimally viable quality criteria.
+This means that these tests do not aim at maximum coverage but instead should test that the most important functionality works.
+This in general is the entire serving infrastructure and the main features of the most used services.
+
+As these tests are executed very often we need them to be as stable, fast and relevant as possible.
+We ensure this by the following criteria:
+
+- Should perfectly conform to all the testing rules laid out [here](../README.md)
+- Does not contain wait times (e.g., for resources to spin up) averaging longer than 20 seconds
+- Should be able to run in parallel. If tests need to run together (e.g., in a scenario), they need to be added to the same test class.
+- Displays some kind of real-world usage. This can be a scenario with multiple services. 
+    - When composing these scenarios, the services should not overlap too much with already existing acceptance tests. We want to avoid redundancy
+    - Existing samples (from [our samples organization](https://github.com/localstack-samples)) can serve as a good starting point for constructing such a scenario
```

Comment:
Is there a specific reason that we highlight the localstack-samples here? Because from our initial efforts on adding scenario tests, we found that most samples are too small and use to little different resources to make great acceptance tests.

Reply:
I think that even though they're quite small, they can still serve as some kind of starting point. Also, ideally we would also like to have samples with some kind of architectural complexity so I think the goals of acceptance tests and samples are kind of aligned here. 

However, I'll add a remark that further complexity might be needed for a sample to become an acceptance test.

Reply:
I think @silv-io suggestion would be sufficient for now :) we also looked at official AWS samples, but here we also have the issue that AWS doesn't keep them up-to-date and they are not guaranteed to still work with AWS, so there might be a lot of additional work required to make them work. But they could at least be used as inspiration for use cases.